### ISSUE TITLE
Next-Gen branch breaks the preconfiguration and initializer

### ISSUE THREAD

**Issue by ychaker** [twilio/twilio-ruby#222]
**Version:** 5.0.0.rc11
**API Subdomain (api/taskrouter/ip_messaging):** Rest Client

### Code Snippet
The `README` on the branch still has the old instructions: 

```ruby
# alternatively, you can preconfigure the client like so
Twilio.configure do |config|
  config.account_sid = account_sid
  config.auth_token = auth_token
end

# and then you can create a new client without parameters
@client = Twilio::REST::Client.new
```

but the code breaks it (I think):
```ruby
def initialize(account_sid, auth_token, http_client=Twilio::HTTP::Client.new)
  ...
end
```

Instead, now we get the following exception:

### Exception / Log
```
ArgumentError:         ArgumentError: wrong number of arguments (given 0, expected 2..3)
```

would be nice to still be able to do the preconfiguration, maybe by setting defaults for those parameters as well.

**Comment by philnash** [comment]
You are right! Thanks for pointing this out. I'll see what I can do to get this updated.

The client file is written by our [auto generating tool](https://www.twilio.com/blog/2016/02/introducing-twilios-next-generation-helper-libraries.html) so it might not be an obvious pull request, but I will work on getting this functionality back in.

**Comment by ychaker** [comment]
I figured that was the case because of the comments at the top of the files.
If I'm not mistaken, the gcloud folks also used to auto generate their code, but then they went away from that because it could not give the users the right feel of using a proper ruby tool. So they are now writing the gems manually. So it's definitely a tough problem to solve if you want to support all the potential communities. 

I actually went to the branch to check if the `README` was updated. I suspected that maybe that method of initializing the client was no longer supported and the `README` would reflect that. So I think if it becomes too hard to get the generated code to maintain that functionality, the `README` should at least be updated so that folks don't go crazy trying to figure what they've done wrong 😄 

Anyway... thank you for your work and your support 🎉

**Comment by ihumanable** [comment]
I think we had decided to not allow using `Twilio::configure` and removed this functionality from the client, then decided there wasn't a great reason to remove this functionality but forgot to add it back in.

It also appears that the client is not enforcing the auth / pathing concern separation that it's supposed to, I will have an update shortly.

**Comment by ihumanable** [comment]
This is fixed in https://github.com/twilio/twilio-ruby/commit/c36bfcd96313735a751fe5e60d964c96003740ee

Released as 5.0.0.RC14
